### PR TITLE
Use sessionid with form calls

### PR DESCRIPTION
**Pull request by Oksamies**

### CODE REVIEW

**Comment by Oksamies** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1030?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #1023** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1023?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1024** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1024?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1029** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1029?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #1030** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1030?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1030?utm_source=stack-comment).

File: packages/thunderstore-api/src/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export interface RequestConfig {
-  apiHost: string;
+  apiHost?: string;
```
Comment (anttimaki) [inline packages/thunderstore-api/src/index.ts:2]: The diff doesn't make it obvious, so why do we want to change the host to be optional? Is some default value provided somewhere else (not visible in this diff)?

File: packages/ts-api-react/src/SessionContext.tsx [hunk 2]
```diff
@@ -49,13 +53,29 @@ const USERNAME_KEY = "username";
  */
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function SessionProvider(props: PropsWithChildren) {
+export function SessionProvider(props: Props) {
   const [_session, _setSession] = useState<LoginResponse>();
   const _storage = new StorageManager("Session");
+
+  // Check if the browser has a Django "sessionid" cookie
+  // Same code is repeated inside useValidateSession
```
Comment (anttimaki) [inline packages/ts-api-react/src/SessionContext.tsx:61]: Why? Is this done upon initialization, and `useValidateSession` can be used later if the user logs in?

File: packages/ts-api-react/src/SessionContext.tsx [hunk 3]
```diff
@@ -147,7 +169,7 @@ const useValidateSession = (
     }
 
     (async () => {
-      const res = await fetch(`${API_DOMAIN}/api/experimental/auth/validate/`, {
+      const res = await fetch(`${domain}/api/experimental/auth/validate/`, {
```
Comment (anttimaki) [inline packages/ts-api-react/src/SessionContext.tsx:172]: Domain can be `undefined`, so this should take it into account somehow.

**Review by anttimaki — COMMENTED** [review]
Hard PR to review since there's so much context that's not visible from the changes. But left some comments anyway.

**Comment by Oksamies** [comment]
### Merge activity

* **Feb 8, 8:22 AM EST**: @Oksamies started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1030).

File: packages/ts-api-react/src/SessionContext.tsx (hunk 2 above)
Comment (Oksamies) [reply packages/ts-api-react/src/SessionContext.tsx:61]: Yep

**Pull request merged**